### ISSUE TITLE
[Feature Request] SMAC Integration

### ISSUE THREAD

**Issue by Benjamin-eecs** [sail-sg/envpool#156]
## Motivation

A typical large-scale multi-agent RL environment for large pretrained model learning

## Resource

https://github.com/oxwhirl/smac

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

**Comment by mavenlin** [comment]
Does this mean translating pysc2 to c++?

**Comment by Benjamin-eecs** [comment]
> https://github.com/oxwhirl/smac

No, actually [SMAC](https://github.com/oxwhirl/smac) is a just python-based multi-agent env based on Blizzard's [StarCraft II Machine Learning API](https://github.com/Blizzard/s2client-proto), which has a [C++](https://github.com/Blizzard/s2client-api) version, [DeepMind](https://deepmind.com/)'s [PySC2](https://github.com/deepmind/pysc2) is another python-based single/multi-agent env different from SMAC, it's famous for [mini games](https://github.com/deepmind/pysc2/releases/download/v1.2/mini_games.zip).
